manifests/init.pp

```diff
@@ -46,47 +47,48 @@
-#   run. E.g. 0 or '00' or [0,30]. Default: fqdn-seeded random minute.
+#   run. E.g. 0 or '00' or [0,30].
 # @param renew_cron_monthday
 #   Optional string, integer or array of monthday(s) the renewal command should
-#   run. E.g. '2-30/2' to run on even days. Default: Every day.
+#   run. E.g. '2-30/2' to run on even days.
 #
 class letsencrypt (
   Optional[String] $email                = undef,
-  String $path                           = $letsencrypt::params::path,
-  $venv_path                             = $letsencrypt::params::venv_path,
+  String $path                           = '/opt/letsencrypt',
+  $venv_path                             = '/opt/letsencrypt/.venv',
   Array $environment                     = [],
-  String $repo                           = $letsencrypt::params::repo,
-  String $version                        = $letsencrypt::params::version,
-  String $package_name                   = $letsencrypt::params::package_name,
-  $package_ensure                        = $letsencrypt::params::package_ensure,
-  String $package_command                = $letsencrypt::params::package_command,
-  String $config_file                    = $letsencrypt::params::config_file,
-  Hash $config                           = $letsencrypt::params::config,
-  String $cron_scripts_path              = $letsencrypt::params::cron_scripts_path,
-  Boolean $manage_config                 = $letsencrypt::params::manage_config,
-  Boolean $manage_install                = $letsencrypt::params::manage_install,
-  Boolean $manage_dependencies           = $letsencrypt::params::manage_dependencies,
-  Boolean $configure_epel                = $letsencrypt::params::configure_epel,
-  Enum['package', 'vcs'] $install_method = $letsencrypt::params::install_method,
-  Boolean $agree_tos                     = $letsencrypt::params::agree_tos,
-  Boolean $unsafe_registration           = $letsencrypt::params::unsafe_registration,
-  Stdlib::Unixpath $config_dir           = $letsencrypt::params::config_dir,
+  String $repo                           = 'https://github.com/certbot/certbot.git',
+  String $version                        = 'v0.30.2',
+  String $package_name                   = 'letsencrypt',
+  $package_ensure                        = 'installed',
+  String $package_command                = 'letsencrypt',
+  Stdlib::Unixpath $config_dir           = '/etc/letsencrypt',
+  String $config_file                    = "${config_dir}/cli.ini",
+  Hash $config                           = {'server' => 'https://acme-v01.api.letsencrypt.org/directory'},
+  String $cron_scripts_path              = "${facts['puppet_vardir']}/letsencrypt",
+  String $cron_owner_group               = 'root',
+  Boolean $manage_config                 = true,
+  Boolean $manage_install                = true,
+  Boolean $manage_dependencies           = true,
+  Boolean $configure_epel                = false,
+  Enum['package', 'vcs'] $install_method = 'vcs',
```

Comment:
All distros we list in metadata.json have it set to `package`. Perhaps that should just be the default? It'd be breaking so I understand not doing it in this PR, but having `Default: 'vcs'` in `REFERENCE.md` is also misleading.

Reply:
i try to reproduce exactly the same it was in params.pp

Reply:
As above: I'd go with `undef` and store all variations in `data/*` for a more understandable `REFERENCE.md`.

Reply:
updated